### PR TITLE
added to do list view and task model

### PR DESCRIPTION
**Pull request by KaitCrawford**
review and merge https://github.com/praekelt/mentorship/pull/5 first

### CODE REVIEW

**Comment by KaitCrawford** [comment]
@miltontony ready for review :)

File: mentorship/mentor/models.py [hunk 1]
```diff
@@ -59,11 +59,30 @@ class ScheduledCall(models.Model):
...
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(blank=True, null=True)
+    mentor = models.ForeignKey('Mentor')
+    description = models.CharField(blank=False,
+                                   null=False,
+                                   max_length=200,
+                                   default="")
+    link = models.URLField(blank=True, null=True, default="")
+    due_date = models.DateField(blank=True, null=True)
+    status = models.PositiveIntegerField(choices=STATUS,
```
Comment (miltontony) [inline mentorship/mentor/models.py]: I feel this should be a CharField with the values `pending, in-progress, done`. A lot easier to look at than 0/1

File: mentorship/mentor/urls.py [hunk 2]
```diff
@@ -22,6 +22,6 @@
 
     url(r'^schedule/delete/(?P<pk>\d+)/$', views.ScheduledCallDelete.as_view(),
         name='delete_schedule'),
-
+    url(r'^task_list/', views.task_list, name='taskList'),
```
Comment (miltontony) [inline mentorship/mentor/urls.py]: naming convention. This should be `task_list`

**Comment by KaitCrawford** [comment]
@miltontony ready for rereview

File: mentorship/mentor/models.py [hunk 3]
```diff
@@ -39,6 +39,9 @@ def dial(self):
     def ended(self):
         self.end_time = datetime.datetime.now()
         self.save()
+        Task.objects.create(
+            mentor=self.caller.mentor,
+            description=("Complete feedback form. Call: %d") % (self.id))
```
Comment (miltontony) [inline mentorship/mentor/models.py]: Probably more useful to have the time in the description than the id..
On that note, should the call be related to the Task via a ForeignKey for future querying?

Comment (miltontony) [inline mentorship/mentor/models.py]: The relationship can be on the Call model e.g `feedback_task`, default empty.

Comment (KaitCrawford) [inline mentorship/mentor/models.py]: Love that idea

**Comment by KaitCrawford** [comment]
@miltontony changes made

File: mentorship/mentor/models.py [hunk 4]
```diff
@@ -39,6 +41,12 @@ def dial(self):
     def ended(self):
         self.end_time = datetime.datetime.now()
         self.save()
+        task = Task.objects.create(
+            mentor=self.caller.mentor,
+            description=("Complete feedback for call at %s on %s") % (
+                self.end_time.strftime('%H:%M'),
+                self.end_time.strftime('%Y-%m-%d')))
+        self.feedback_task = task
```
Comment (miltontony) [inline mentorship/mentor/models.py]: I think you need to save afterwards. Perhaps move line 43 to after the task is assigned

Comment (miltontony) [inline mentorship/mentor/models.py]: On that note...I think we should enable travis. I think this tasks description should have a test

Comment (KaitCrawford) [inline mentorship/mentor/models.py]: It's tested as part of test_end_call() https://github.com/praekelt/mentorship/pull/6/files#diff-4ec985fb4eb45f3d4422b54b114d0e52R100
but +1 on enabling travis, I just have no idea how to do that :grin:

**Comment by KaitCrawford** [comment]
@miltontony travis done and other feedback implemented

**Comment by miltontony** [comment]
:+1:

**Pull request merged**